core/src/main/java/com/linecorp/armeria/client/BlockingResponseAs.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.util.Exceptions;
+
+public class BlockingResponseAs implements ResponseAs<HttpResponse, AggregatedHttpResponse> {
```

Comment:
Sorry about the misdirection, but what do you think of moving the methods here to overload methods in `ResponseAs.java`?

```java
static <T> BlockingConditionalResponseAs<T> json(Class<? extends T> clazz,
                                                 Predicate<AggregatedHttpResponse> predicate) {
    requireNonNull(clazz, "clazz");
    return new BlockingConditionalResponseAs<>(ResponseAs.blocking(),
                                               AggregatedResponseAs.json(clazz), predicate);
}
```

And then users can just do this

```java
WebClient.of(server.httpUri()).prepare().get("/json_generic_server_error")
         .as(ResponseAs.json(new TypeReference<List<MyObject>>() {},
                             res -> res.status().isServerError())
```

By doing so, I don't think users have to type out `blocking()` and just chain conditions immediately

Reply:
I think it's good because users can do the same thing more simply. Thanks!